tests/common/beamlines/test_device_factory.py

```diff
@@ -0,0 +1,90 @@
+from bluesky.run_engine import RunEngine
+from ophyd_async.epics.motion.motor import Motor
+from ophyd_async.plan_stubs.ensure_connected import ensure_connected
+
+from dodal.common.beamlines.beamline_utils import ACTIVE_DEVICES
+from dodal.common.beamlines.device_factory import (
+    DeviceInitializationConfig,
+    device_factory,
+)
+
+
+def test_terminal_use_case_decorated_motor_not_mock(RE: RunEngine):
+    @device_factory()
+    def motor(name: str = "motor", prefix: str = "motor:"):
+        return Motor(name=name, prefix=prefix)
+
+    m = motor(name="foo", mock=True)
+
+    assert m is not None
+    assert m.name == "foo"
+
+    RE(ensure_connected(m))
```

Comment:
Is there a better way to test that a device has been connected? /is mock?

Reply:
that's the recommended way, specifically designed to work in conjunction with this PR. what is `not-better` about this plan stub?

Reply:
https://github.com/bluesky/ophyd-async/blob/0a1b6611f259296beadff955b94f3311ed2c9689/src/ophyd_async/plan_stubs/_ensure_connected.py#L6

ensure_connected takes a bunch of devices and calls connect on them. That's not the same as asserting that they are connected.

Reply:
From looking at Signal and SignalBackend, there's no real way to check if something has had connect called on it.

This seems like a use case where we're required to make some MockDevice class, and assert_called on its connect

Reply:
Yes, at runtime the idea is that `ensure_connected` should be cheap enough that you can call it every time you need a connected signal. This leaves us free in the backend to do some intelligent caching in the future that might invalidate that cache when a single PV from that IOC disconnects.

For tests then there is no public way to tell that connect has ever been run. If this is required we could make a `last_connect: Task` property that would tell you.